weasyl/journal.py

```diff
@@ -106,14 +106,53 @@ def select_view(userid, rating, journalid, ignore=True, anyway=None):
         "favorited": favorite.check(userid, journalid=journalid),
         "friends_only": "f" in journal.settings,
         "hidden_submission": "h" in journal.settings,
-        # todo
-        "fave_count": d.execute("SELECT COUNT(*) FROM favorite WHERE (targetid, type) = (%i, 'j')",
-                                [journalid], ["element"]),
+        "fave_count": favorite.count(journalid=journalid),
         "tags": searchtag.select(journalid=journalid),
         "comments": comment.select(userid, journalid=journalid),
     }
 
 
+def select_view_api(userid, journalid, anyway=False, increment_views=False):
+    rating = d.get_rating(userid)
+    db = d.connect()
+    journal = db.query(orm.Journal).get(journalid)
+    if journal is None or 'hidden' in journal.settings:
+        raise WeasylError('journalRecordMissing')
+    journal_rating = journal.rating.code
+    if 'friends-only' in journal.settings and not frienduser.check(userid, journal.userid):
+        raise WeasylError('journalRecordMissing')
+    elif journal_rating > rating and userid != journal.userid:
+        raise WeasylError('RatingExceeded')
+    elif not anyway and ignoreuser.check(userid, journal.userid):
+        raise WeasylError('UserIgnored')
+    elif not anyway and blocktag.check(userid, journalid=journalid):
+        raise WeasylError('TagBlocked')
+
+    views = journal.page_views
+    if increment_views and d.common_view_content(userid, journalid, 'journal'):
+        views += 1
+
+    content = files.read(files.make_resource(userid, journalid, 'journal/submit'))
+
+    return {
+        'journalid': journalid,
+        'title': journal.title,
+        'owner': journal.owner.profile.username,
+        'owner_login': journal.owner.login_name,
+        'content': text.markdown(content),
+        'tags': searchtag.select(journalid=journalid),
+        'link': d.absolutify_url('/journal/%d/%s' % (journalid, text.slug_for(journal.title))),
+
+        'type': 'journal',
+        'rating': journal.rating.name,
+
+        'views': views,
+        'favorites': favorite.count(journalid=journalid),
+        'comments': comment.count(journalid=journalid),
+        'favorited': favorite.check(userid, journalid=journalid),
+        'friends_only': 'friends-only' in journal.settings
```

Comment:
Yeah, sorry about that. I started one on the wiki: https://github.com/Weasyl/weasyl/wiki/Style-guide